### PR TITLE
use LC_ALL=C on all sort commands

### PR DESCRIPTION
**Pull request by eparis**
Most of these don't matter, but this makes it predicable no matter who runs the commands.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35221)

### CODE REVIEW

**Comment by eparis** [comment]
@thockin you looked at the last of these, this should be pretty easy
(I cheated and cleaned up some golang 1.7 lint things too)

File: cmd/genyaml/gen_kubectl_yaml.go [hunk 1]
```diff
@@ -83,7 +83,7 @@ func forceMultiLine(s string) string {
 }
 
 func genFlagResult(flags *pflag.FlagSet) []cmdOption {
-	result := make([]cmdOption, 0)
+	var result []cmdOption
```
Comment (thockin) [inline cmd/genyaml/gen_kubectl_yaml.go]: unrelated

File: plugin/pkg/auth/authenticator/request/x509/x509_test.go [hunk 2]
```diff
@@ -765,7 +765,7 @@ func getCert(t *testing.T, pemData string) *x509.Certificate {
 }
 
 func getCerts(t *testing.T, pemData ...string) []*x509.Certificate {
-	certs := make([]*x509.Certificate, 0)
+	var certs []*x509.Certificate
```
Comment (thockin) [inline plugin/pkg/auth/authenticator/request/x509/x509_test.go]: unrelated

**Comment by eparis** [comment]
those were the golang 1.7 lint complaints. I can back them out. One second.

**Comment by eparis** [comment]
and cleanups the break 1.7.1. lint gone. Will send those in another PR.
